tests/test_frontdoor.py

```diff
@@ -2010,6 +2010,65 @@ def test_on_epc_select_page_manual_link_is_shown_if_no_addresses_found():
     assert page.has_one('h1:contains("What is the property\'s address?")')
 
 
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockEPCApi)
+def test_epc_page_shows_epc_info():
+    client = utils.get_client()
+    page = client.get("/start")
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.status_code == 200
+    session_id = page.path.split("/")[1]
+    assert uuid.UUID(session_id)
+    _check_page = _make_check_page(session_id)
+
+    form = page.get_form()
+    form["country"] = "England"
+    page = form.submit().follow()
+
+    form = page.get_form()
+    form["supplier"] = "Utilita"
+    page = form.submit().follow()
+
+    assert page.has_text("Do you own the property?")
+    page = _check_page(page, "own-property", "own_property", "Yes, I own my property and live in it")
+
+    assert page.has_text("Do you live in a park home")
+    page = _check_page(page, "park-home", "park_home", "No")
+
+    form = page.get_form()
+    form["building_name_or_number"] = "22"
+    form["postcode"] = "FL23 4JA"
+    page = form.submit().follow()
+
+    form = page.get_form()
+    form["lmk"] = "222222222222222222222222222222222"
+    page = form.submit().follow()
+
+    data = interface.api.session.get_answer(session_id, page_name="epc-select")
+    assert data["lmk"] == "222222222222222222222222222222222"
+    assert data["address"] == "22 Acacia Avenue, Upper Wellgood, Fulchester, FL23 4JA"
+
+    assert page.has_one("h1:contains('What is the council tax band of your property?')")
+    page = _check_page(page, "council-tax-band", "council_tax_band", "B")
+
+    assert page.has_one("h1:contains('We found an Energy Performance Certificate that might be yours')")
+    assert page.has_one("p:contains('Registered address')")
+    assert page.has_one("p:contains('22 Acacia Avenue')")
+    assert page.has_one("p:contains('Upper Wellgood')")
+    assert page.has_one("p:contains('Fulchester')")
+    assert page.has_one("p:contains('FL23 4JA')")
+
+    assert page.has_one("p:contains('Property type')")
+    assert page.has_one("p:contains('Maisonette')")
+
+    assert page.has_one("p:contains('EPC rating')")
+    assert page.has_one("p:contains('G')")
+
+    assert page.has_one("p:contains('Date of issue')")
+    assert page.has_one("p:contains('23 July 2010')")
+
+
```

Comment:
would like to have added more tests for the english & welsh discrepancy, but testino struggles with setting a language. had to abandon those tests for now, though kept some of the refactors to test EPC loading